src/Cake.Common/IO/FileCopier.cs

```diff
@@ -128,5 +148,35 @@ private static void CopyFileCore(ICakeContext context, FilePath filePath, FilePa
             var file = context.FileSystem.GetFile(absoluteFilePath);
             file.Copy(absoluteTargetPath, true);
         }
+
+        private static void CopyFileCore(ICakeContext context, FilePath filePath, FilePath targetFilePath, string commonPath, string targetDirectoryPath)
+        {
+            var absoluteFilePath = filePath.MakeAbsolute(context.Environment);
+
+            // Get the file.
+            if (!context.FileSystem.Exist(absoluteFilePath))
+            {
+                const string format = "The file '{0}' do not exist.";
+                var message = string.Format(CultureInfo.InvariantCulture, format, absoluteFilePath.FullPath);
+                throw new FileNotFoundException(message, absoluteFilePath.FullPath);
+            }
+
+            // Copy the file.
+            var absoluteTargetPath = targetFilePath.MakeAbsolute(context.Environment);
+            var file = context.FileSystem.GetFile(absoluteFilePath);
+
+            // Get the parent folder structure and create it.
+            var newRelativeFolderPath = filePath.GetDirectory().ToString().Replace(commonPath.Replace("\\", "/"), string.Empty);
+            var newTargetPath = absoluteTargetPath.GetDirectory() + newRelativeFolderPath;
```

Comment:
Try switching the values from `first.GetRelativePath(second)` to `second.GetRelativePath(first)`.